docs/aws/bin_packing.md

```diff
@@ -0,0 +1,15 @@
+# Bin packing
+
+Karpenter provisions instances based on the number of pending pods and their resource requirements (CPU and Memory). These requirements vary and is configured by the pod owners. Karpenter needs to make sure that there is sufficient capacity among the provisioned instances for these pods to be scheduled.
+
+The easiest way is to provision an instance per pending pod, however, its not very efficient and can incur infrastructure cost for the user. In order to be able to schedule pods efficiently and be cost effective, Karpenter packs these pods on to the available instance types with the given set of constraints. Karpenter follows the [First Fit Decreasing](https://en.wikipedia.org/wiki/Bin_packing_problem#First_Fit_Decreasing_(FFD)) algorithm for bin packing the pods.
```

Comment:
Could you add a brief sentence on why we chose FFD (for now?) - it's great there's a link to the algorithm but it would be also great to explicitly say why we chose it (otherwise the reader has to go to the doc and then kind of guess why we liked it?)